content/blog/2019-05-14-nasapower.md

```diff
@@ -0,0 +1,625 @@
+gridded data of a thirty year time period from January 1984 to December
+2013 for the whole global surface.
+
+The CLIMATOLOGY data are the only way to get the entire surface in one
+go as shown here for the global annual average temperature, T2M.
+
+``` r
+library(raster)
+```
+
+    ## Loading required package: sp
+
+    ## 
+    ## Attaching package: 'raster'
+
+    ## The following object is masked from 'package:tidyr':
+    ## 
+    ##     extract
+
+``` r
+library(viridisLite)
+
+global_t2m <-
+  get_power(
+    community = "AG",
+    pars = "T2M",
+    temporal_average = "CLIMATOLOGY",
+    lonlat = "GLOBAL"
+  )
+
+# view the tibble
+global_t2m
+```
+
+    ## NASA/POWER SRB/FLASHFlux/MERRA2/GEOS 5.12.4 (FP-IT) 0.5 x 0.5 Degree Climatologies  
+    ##  22-year Additional Solar Parameter Monthly & Annual Climatologies (July 1983 - June 2005), 30-year Meteorological and Solar Monthly & Annual Climatologies (January 1984 - December 2013)  
+    ##  Location: Global  
+    ##  Value for missing model data cannot be computed or out of model availability range: -99  
+    ##  Parameter(s):  
+    ##  T2M MERRA2 1/2x1/2 Temperature at 2 Meters (C)  
+    ##  
+    ##  Parameters: 
+    ##  NA;
+    ##  NA;
+    ##  T2M MERRA2 1/2x1/2 Temperature at 2 Meters (C)  
+    ##  
+    ## # A tibble: 259,200 x 16
+    ##      LON   LAT PARAMETER   JAN   FEB   MAR   APR   MAY   JUN   JUL   AUG
+    ##    <dbl> <dbl> <chr>     <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
+    ##  1 -180. -89.8 T2M       -29.0 -40.7 -52.9 -57.8 -59.1 -59.6 -61.3 -61.8
+    ##  2 -179. -89.8 T2M       -29.0 -40.7 -52.9 -57.8 -59.1 -59.6 -61.3 -61.8
+    ##  3 -179. -89.8 T2M       -29.0 -40.7 -52.9 -57.8 -59.1 -59.6 -61.3 -61.8
+    ##  4 -178. -89.8 T2M       -29.0 -40.7 -52.9 -57.8 -59.1 -59.6 -61.3 -61.8
+    ##  5 -178. -89.8 T2M       -29.0 -40.7 -52.9 -57.8 -59.1 -59.6 -61.3 -61.8
+    ##  6 -177. -89.8 T2M       -28.9 -40.7 -52.9 -57.9 -59.1 -59.6 -61.3 -61.8
+    ##  7 -177. -89.8 T2M       -28.9 -40.7 -52.9 -57.9 -59.1 -59.6 -61.3 -61.8
+    ##  8 -176. -89.8 T2M       -28.9 -40.7 -53.0 -57.9 -59.1 -59.6 -61.3 -61.8
+    ##  9 -176. -89.8 T2M       -28.9 -40.7 -53.0 -57.9 -59.1 -59.6 -61.3 -61.8
+    ## 10 -175. -89.8 T2M       -28.9 -40.7 -53.0 -57.9 -59.1 -59.6 -61.3 -61.8
+    ## # … with 259,190 more rows, and 5 more variables: SEP <dbl>, OCT <dbl>,
+    ## #   NOV <dbl>, DEC <dbl>, ANN <dbl>
+
+``` r
+# map only annual average temperatures by converting the 15th column to a raster
+# object
+T2M_ann <- rasterFromXYZ(
+  global_t2m[c(1:2, 16)],
+  crs = "+proj=eqc +lat_ts=0 +lat_0=0 +lon_0=0 +x_0=0 +y_0=0 +ellps=WGS84 +datum=WGS84 +units=m +no_defs")
+
+# how many unique values
+n <- length(unique(global_t2m$ANN))
+
+# plot the annual average temperature using viridis
+plot(T2M_ann, col = viridis(n = n))
+```
+
+![](/img/blog-images/2019-05-14-nasapower/global-T2M-1.png)<!-- -->
+
+Otherwise you have to specify a single cell or regional coverage that
```

Comment:
Can you give one short line explaining this global temp fig and the following south-western corner of Australia fig since there's a fair bit of code between your preamble and the figs?